robocjk/logging.py

```diff
@@ -0,0 +1,32 @@
+import logging
+import traceback
+
+
+class ThrottleFilter(logging.Filter):
+    def __init__(self, timeout=3600):
+        super().__init__()
+        self.timeout = timeout
+
+    def filter(self, record):
+        try:
+            from django.core.cache import caches
+
+            cache = caches["logging"]
+            if record.exc_info:
+                exc_type = record.exc_info[0]
+                exc_value = record.exc_info[1]
+                exc_traceback = exc_value.__traceback__
+                lastframe = traceback.extract_tb(exc_traceback)[-1]
+                key = (
+                    "logging.ThrottleFilter "
+                    f"{exc_type.__name__} "
+                    f"{lastframe.filename} "
+                    f"{lastframe.lineno} "
+                    f"{lastframe.name}"
+                )
+                if cache.get(key):
+                    return False
+                cache.set(key, True, self.timeout)
+        except Exception:
```

Comment:
Are you sure you want to silence everything unexpected? That may lead to hard to debug problems.

Reply:
Ah I see what this does: if anything fails here, you'll get the 600 emails per hour after all, right? :)

Reply:
The try block contains only the logic for the throttling, if something unexpected happens there, the filter will return `True` and the email will be sent bypassing the throttling logic.

If the filter returns `True` the email will be sent, if the filter returns `False`, the email will be not sent.